### ISSUE TITLE
Consumer (1m) / Producer (1d) does not pass data or only once per 24h?

### ISSUE THREAD

**Issue by enshoe** [freqtrade/freqtrade#9203]
## Describe your environment

  * Operating system: DEBIAN GNU/LINUX 10
  * Python Version: 2.7.1.6
  * CCXT version: 4.0.71
  * Freqtrade Version: 2023.8.Stable
  
## Describe the problem:
In my main strategy (consumer, 1m) I want to add 20, 50, 100 & 200 day EMA's. In order to do this I created a support producer on a 1d timeframe with the required indicators. I can see the them in the consumer dataframe, correctly named, but they hold no data.

Does the data only get sent once, on generation every 24 hours? If so, is that intended?

### Steps to reproduce:

  1. Create consumer 1m timeframe
  2. Create producer 1d timeframe
  3. Entries exist on consumer side, but holds no data.
  
### Observed Results:

  * What happened?
"-" being shown on consumer under ema20_five

  * What did you expect to happen?
ema20_five to show the current ema20 from the 1d timeframe

### Relevant code exceptions or logs
  ```
Consumer log example (Constant spam):
2023-09-18 19:53:38,641 - freqtrade.rpc.external_message_consumer - WARNING - Holes in data or no existing df, requesting 1500 candles for ('CYBER/BTC', '1d', <CandleType.SPOT: 'spot'>) from `five`

Producer log example (Constant spam):
023-09-18 19:54:07,660 - freqtrade.rpc.api_server.api_ws - DEBUG - Request of type analyzed_df from WebSocketChannel(XXXX, ('172.22.0.7', 52852))
2023-09-18 19:54:07,661 - uvicorn.error - DEBUG - > TEXT '{"type":"analyzed_df","data":{"key":["SEI/BTC",...9:49:44.017090+00:00"}}' [4283 bytes]
  ```

Consumer strategy file:
```

    _producers = ['one', 'two', 'three', 'four', 'five']
    _producer_tfs = {
        'one': '5m',
        'two': '1m',
        'three': '5m',
        'four': '5m',
        'five': '1d'
    }

    _columns_to_expect = {}

    for producer in _producers:
        _columns_to_expect[producer] = [
            f'enter_long_{producer}',
            f'enter_long_{producer}',
            f'enter_long_{producer}',
            f'enter_long_{producer}',
            f'enter_long_{producer}'
        ]

    def populate_indicators(self, df: DataFrame, metadata: dict) -> DataFrame:
        pair = metadata['pair']
        timeframe = self.timeframe
        producer_pairs = self.dp.get_producer_pairs()

        for producer in self._producers:

            producer_timeframe = self._producer_tfs[producer]
            producer_df, _ = self.dp.get_producer_df(pair, timeframe=producer_timeframe, producer_name=producer)

            if not producer_df.empty:

                df = merge_informative_pair(df, producer_df, timeframe,
                                                    producer_timeframe,
                                                    append_timeframe=False,
                                                    suffix=producer, ffill=False)
            else:
                df[self._columns_to_expect[producer]] = np.nan

        return df
  ```

Producer populate indicators:
```
    def populate_indicators(self, dataframe: DataFrame, metadata: dict) -> DataFrame:
        ## Add indicators below
        dataframe['ema20'] = ta.EMA(dataframe, timeperiod=20)
        dataframe['ema50'] = ta.EMA(dataframe, timeperiod=50)
        dataframe['ema100'] = ta.EMA(dataframe, timeperiod=100)
        dataframe['ema200'] = ta.EMA(dataframe, timeperiod=200)

        return dataframe
```

**Comment by enshoe** [comment]
![Screenshot_20230919_025801_Chrome](https://github.com/freqtrade/freqtrade/assets/27415969/6cb9beca-1439-49af-b030-b3000b02f497)

**Update:**
The ema shows up once per 24 hours. 

The constant spam in the logs about holes in the data or missing dataframe could probably be handled better in this specific case.

**Comment by xmatthias** [comment]
I don't think your problem is the 1d data - but it's a general problem of "data does not fit together" as the distance between 1m and 1d is too big.

by default, you'll get ~1000 closed candles from the exchange (on both 1d and 1m timeframes).

today (19th sept) - the very last closed 1d candle is the one from 18th (2023-09-18 00:00:00 -> 2023-09-19 00:00:00 - with the candle having the date of 2023-09-18 00:00:00).
to successfully merge this candle, you'll therefore need to have at least 2 days of 1m candles (which you won't have on 1m timeframe) - with one date being at least 2023-09-18 00:00:00 (otherwise you also can't ffill - as you have no overlapping points).

This is a common problem if the gap between the short and long timeframe is too big - and the only real solution is to use smaller gap.

---

as for the spam - that's most likely because you'll not have sufficient 1d candles (the base assumption is of 100 candles - which i guess CYBER is not being listed at so far).

The bigger question is - why are you using producer/consumers in the first place? 
it seems like a regular strategy doing some light work (calculating EMA) - so your setup (using multiple producers) seesm way over top.
you can accomplish the same thing with one running bot process (and most likely also more efficiently) with [`@informative`](https://www.freqtrade.io/en/latest/strategy-customization/#informative-pairs-decorator-informative).
This won't fix the "the gap between candles is too wide" problem - but will greatly simplify your deployment, as well as the strategy development.

producer/consumer should be considered as advanced setup - when doing the work in one process / on one system is not feasible - which (with the information given above) - really don't see as being the case.

**Comment by enshoe** [comment]
I am using several bots (producers) which generally have a high workload (many pairs, different timeframes). They generate signals, which I then filter on my consumer and do the actual trading.

I was unable to get the 20, 50, 100 & 200 1d EMA's into my 1m consumer strategy reliably. The points were only generated once per 24h, potentially giving my consumer a very long startup time.

Now I have created a very simple producer on 4h timeframe, which in itself have a 1d informative. This gives me several 1d datapoints to use in my 1m consumer strategy, readily available on startup.
![image](https://github.com/freqtrade/freqtrade/assets/27415969/020a5130-d254-443a-b2bb-896fe4f4924e)

**Comment by xmatthias** [comment]
> I was unable to get the 20, 50, 100 & 200 1d EMA's into my 1m consumer strategy reliably. The points were only generated once per 24h, potentially giving my consumer a very long startup time.

this is wrong - **updates** to 1d candles will only happen once per day (once per candle) - but you'll get initial startup values just fine (these are done as explicit request).
This obviously assumes that data for this pair is available on the producer when you start the consumer.

Your problem will happen on merges, where (depending on the startup time) you'll have no corresponding 1m candle - and potentially on pairs that are not listed for this long (like, less than 100 days).
You'll notice that your 1m chart doesn't start prior to yesterday - so there's no common datapoint with the candle (as mentioned above, the candle date for the current time is sept 18th - which means it'll only be merged to that minute's candle - which is not available.

on 4h candles you'll not really have this problem - (unless you're on okx) - as 4h candles happen more frequently.
The Screenshot also confirms this to some degree - as you've got data every 4 hours. This also tells me that you ffill the 1d data on 4h candles - so the data will be an "even line" throughout all candles within the day (which is common behavior).
